erizo/src/erizo/WebRtcConnection.cpp

```diff
@@ -400,11 +401,25 @@ int WebRtcConnection::deliverFeedback_(char* buf, int len) {
   return len;
 }
 
-void WebRtcConnection::onTransportData(char* buf, int len, Transport *transport) {
+void WebRtcConnection::onTransportData(std::shared_ptr<dataPacket> packet, Transport *transport) {
   if (audioSink_ == NULL && videoSink_ == NULL && fbSink_ == NULL) {
     return;
   }
 
+  packetType type = OTHER_PACKET;
+
+  if (transport->mediaType == AUDIO_TYPE) {
+    type = AUDIO_PACKET;
+  } else if (transport->mediaType == VIDEO_TYPE) {
+    type = VIDEO_PACKET;
+  }
+
+  inbound_pipeline_->read(packet);
```

Comment:
The packet here is a shared_ptr that points to a packet that will change for every packet received. Meanwhile, in the output pipeline, we have a newly allocated shared_ptr for each of the packets in the pipeline. 
As far as I can tell, this works here but will create confusion when we start implementing more pipeline handlers. I would probably make a copy to make it clearer, at the expense of an extra copy that is not really needed right now. Again, it works fine as it is as far as I can tell and it could wait until we clean up more stuff